ols/app/endpoints/ols.py

```diff
@@ -184,8 +184,14 @@ def generate_response(
         docs_summarizer = DocsSummarizer(
             provider=llm_request.provider, model=llm_request.model
         )
+        # Note: This may change, if we change how we store history currently.
```

Comment:
I'd probably refactor it, or defer this work to `DocsSummarizer`. Then the change in API logic will be minimal

Reply:
well yes, in the end it will make the cache content "readable". I'm just thinking there we might focus on short-term solution, to fix the token limit problem first. Then (after internal release+EA) we'll have time to refactor everything else.